code/lib/cli/src/js-package-manager/JsPackageManager.ts

```diff
@@ -76,6 +76,49 @@ export abstract class JsPackageManager {
     this.cwd = options?.cwd || process.cwd();
   }
 
+  /** Detect whether Storybook gets initialized in a monorepository/workspace environment
+   * The cwd doesn't have to be the root of the monorepo, it can be a subdirectory
+   * @returns true, if Storybook is initialized inside a monorepository/workspace
+   */
+  public isStorybookInMonorepo() {
+    let cwd = process.cwd();
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      try {
+        const turboJsonPath = `${cwd}/turbo.json`;
```

Comment:
should it check for nx.json, rush.json and lerna.json as well?

Reply:
Good point for `rush.json`. I haven't it on my radar. Though, it is not necessary for `nx` or `lerna`. First of all, the `nx.json` is optional. Second, `nx` and `lerna` both rely on the `workspaces` field set in the `package.json`. Therefore it is not necessary to do further read operations on those configuration files.